src/rules/prefer-constructor.ts

```diff
@@ -0,0 +1,150 @@
+import {
+  AST_NODE_TYPES,
+  TSESTree,
+} from "@typescript-eslint/experimental-utils";
+import { constant, constVoid, flow, pipe } from 'fp-ts/function';
+import { boolean, option, readonlyNonEmptyArray } from "fp-ts";
+import {
+  calleeIdentifier,
+  contextUtils,
+  createRule
+} from '../utils';
+
+export default createRule({
+  name: "prefer-constructor",
+  meta: {
+    type: "suggestion",
+    fixable: "code",
+    schema: [],
+    docs: {
+      category: "Best Practices",
+      description: "afsafaf",
+      recommended: "warn",
+    },
+    messages: {
+      eitherFoldIsOptionFromEither: "cacsaffg",
+      replaceEitherFoldWithOptionFromEither: "dsagdgdsg",
+    },
+  },
+  defaultOptions: [],
+  create(context) {
+    const { isIdentifierImportedFrom } = contextUtils(context);
+    return {
+      CallExpression(node) {
+        const isEitherFold = (node: TSESTree.CallExpression) => {
+            return pipe(
+              node,
+              calleeIdentifier,
+              option.filter((callee) => callee.name === "fold"),
+              option.chain(flow((callee) => callee.parent, option.fromNullable)),
+              option.filter((parent): parent is TSESTree.MemberExpression => parent.type === AST_NODE_TYPES.MemberExpression),
+              option.map((parent) => parent.object),
+              option.filter((object): object is TSESTree.Identifier => object.type === AST_NODE_TYPES.Identifier),
+              option.exists((object) => object.name === 'either')
```

Comment:
Currently looking for a way to determine if it's `either.fold`, even if it's `E.fold` or whatever.

Reply:
Two ways:

- use the type information (more accurate, but slower)
- determine whether the "prefix" is imported from `fp-ts` (accurate enough, and faster because it's only a syntactic lookup)

Reply:
I'll take a closer look as soon as I have minute